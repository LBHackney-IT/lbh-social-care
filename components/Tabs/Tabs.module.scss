@import 'lbh-frontend/lbh/base';

.tabList {
  border-bottom: 1px solid lbh-colour('lbh-border');
  margin-bottom: 1rem;
}

.tab {
  margin-top: 0;
  display: inline-block;
  margin-right: 20px;
  padding-bottom: 10px;
  &:last-of-type {
    margin-right: 0;
  }
}

.link {
  text-decoration: none;
  color: lbh-colour('lbh-link');
}

.active {
  a {
    font-weight: 600;
  }
  border-bottom: 4px solid lbh-colour('lbh-link');
}

.panel {
  border: none;
  padding: 0;
}
