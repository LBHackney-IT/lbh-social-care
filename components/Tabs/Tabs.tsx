import cx from 'classnames';
import Link from 'next/link';
import { useRouter } from 'next/router';

interface Tab {
  url: string;
  text: string;
}

export interface Props {
  title: string;
  tabs: Tab[];
  children: React.ReactChild;
}

const Tabs = ({ title, tabs, children }: Props): React.ReactElement => {
  const { pathname, query } = useRouter();
  return (
    <div className="govuk-tabs">
      <h2 className="govuk-tabs__title">{title}</h2>
      <ul className="govuk-tabs__list">
        {tabs.map(({ url, text }) => (
          <li
            key={text}
            className={cx('govuk-tabs__list-item', {
              'govuk-tabs__list-item--selected': pathname === url,
            })}
          >
            <Link href={{ pathname: url, query: query }} scroll={false}>
              <a className="govuk-tabs__tab">{text}</a>
            </Link>
          </li>
        ))}
      </ul>
      <div className="govuk-tabs__panel">{children}</div>
    </div>
  );
};

export default Tabs;
