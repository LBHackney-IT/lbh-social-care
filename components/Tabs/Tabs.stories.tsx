import { Story } from '@storybook/react';

import Tabs, { Props } from './Tabs';

export default {
  title: 'Tabs',
  component: Tabs,
};

const Template: Story<Props> = (args) => (
  <div className="js-enabled">
    <Tabs {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  title: 'foo',
  tabs: [
    {
      url: '/foo',
      text: 'foo',
    },
    { url: '/bar', text: 'bar' },
  ],
  children: <p>I am the content!</p>,
};
Default.parameters = {
  nextRouter: {
    pathname: '/foo',
  },
};

export const SecondSelected = Template.bind({});
SecondSelected.args = {
  title: 'foo',
  tabs: [
    {
      url: '/foo',
      text: 'foo',
    },
    { url: '/bar', text: 'bar' },
  ],
  children: <p>I am the content!</p>,
};
SecondSelected.parameters = {
  nextRouter: {
    pathname: '/bar',
  },
};
