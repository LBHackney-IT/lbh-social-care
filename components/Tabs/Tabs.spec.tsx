import { render } from '@testing-library/react';

import Tabs from './Tabs';

describe(`Tabs`, () => {
  const props = {
    title: 'foo',
    tabs: [
      {
        url: '/foo',
        text: 'foo',
        isSelected: true,
      },
      { url: '/bar', text: 'bar' },
    ],
    children: <p>I am the content!</p>,
  };

  it('should render properly', () => {
    const { asFragment } = render(<Tabs {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
