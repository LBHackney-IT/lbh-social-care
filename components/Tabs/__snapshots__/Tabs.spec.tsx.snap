// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render properly 1`] = `
<DocumentFragment>
  <div
    class="lbh-tabs govuk-tabs"
  >
    <h2
      class="govuk-tabs__title"
    >
      foo
    </h2>
    <ul
      class="govuk-tabs__list"
    >
      <li
        class="lbh-tabs govuk-tabs__list-item govuk-tabs__list-item--selected"
      >
        <a
          class="lbh-tabs govuk-tabs__tab"
          href="/foo"
        >
          foo
        </a>
      </li>
      <li
        class="lbh-tabs govuk-tabs__list-item"
      >
        <a
          class="lbh-tabs govuk-tabs__tab"
          href="/bar"
        >
          bar
        </a>
      </li>
    </ul>
    <div
      class="lbh-tabs govuk-tabs__panel"
    >
      <p>
        I am the content!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
